################################################################################
# This patch file is from Qualcomm Innovation Center, Inc. and is provided under
# the following license:
# Copyright (c) 2025 Qualcomm Innovation Center, Inc. All rights reserved.
# SPDX-License-Identifier: BSD-3-Clause-Clear
################################################################################
From 6efdb364a6d17f137df33f689e0d733f458d1c3e Mon Sep 17 00:00:00 2001
From: Tushar Darote <quic_tdarote@quicinc.com>
Date: Tue, 10 Dec 2024 18:10:36 +0530
Subject: [PATCH] Tensorflow-lite: Rename of
 cl_ndrange_kernel_command_properties_khr to cl_command_properties_khr

Change-Id: Ic4dcad80c3844858267411d9e37408e0ac0922cd
Signed-off-by: Tushar Darote <quic_tdarote@quicinc.com>
---
 tensorflow/lite/delegates/gpu/cl/opencl_wrapper.h | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/tensorflow/lite/delegates/gpu/cl/opencl_wrapper.h b/tensorflow/lite/delegates/gpu/cl/opencl_wrapper.h
index f946e8e29c0..b2b75ff783f 100644
--- a/tensorflow/lite/delegates/gpu/cl/opencl_wrapper.h
+++ b/tensorflow/lite/delegates/gpu/cl/opencl_wrapper.h
@@ -543,6 +543,19 @@ typedef cl_int(CL_API_CALL *PFN_clEnqueueCommandBufferKHR)(
     cl_uint /*num_events_in_wait_list*/, const cl_event * /*event_wait_list*/,
     cl_event * /*event*/);

+#if CL_KHR_COMMAND_BUFFER_EXTENSION_VERSION >= CL_MAKE_VERSION(0, 9, 5)
+typedef cl_int(CL_API_CALL *PFN_clCommandNDRangeKernelKHR)(
+    cl_command_buffer_khr /*command_buffer*/,
+    cl_command_queue /*command_queue*/,
+    const cl_command_properties_khr * /*properties*/,
+    cl_kernel /*kernel*/, cl_uint /*work_dim*/,
+    const size_t * /*global_work_offset*/, const size_t * /*global_work_size*/,
+    const size_t * /*local_work_size*/,
+    cl_uint /*num_sync_points_in_wait_list*/,
+    const cl_sync_point_khr * /*sync_point_wait_list*/,
+    cl_sync_point_khr * /*sync_point*/,
+    cl_mutable_command_khr * /*mutable_handle*/);
+#else
 typedef cl_int(CL_API_CALL *PFN_clCommandNDRangeKernelKHR)(
     cl_command_buffer_khr /*command_buffer*/,
     cl_command_queue /*command_queue*/,
@@ -554,6 +567,7 @@ typedef cl_int(CL_API_CALL *PFN_clCommandNDRangeKernelKHR)(
     const cl_sync_point_khr * /*sync_point_wait_list*/,
     cl_sync_point_khr * /*sync_point*/,
     cl_mutable_command_khr * /*mutable_handle*/);
+#endif

 typedef cl_int(CL_API_CALL *PFN_clGetCommandBufferInfoKHR)(
     cl_command_buffer_khr /*command_buffer*/,
--
2.25.1

